_posts/2024-04-29-dependabot-management.markdown

```diff
@@ -0,0 +1,206 @@
+
+## AWS Code Artifact
+
+<center>
+<div ><img src="/img/dependabot-management/code_artifact_issue.png" width="400px"/><p>Image 2: AWS CodeArtifact</p></div>
+</center>
+
+Another common issue encountered with Dependabot is the inability to authenticate with AWS CodeArtifact at a organization level (to my knowledge). AWS Code Artifact is used for a variety of reasons including storing private packages and sharing them across your organization. In the context Dependabot updates, this can cause an issue as it is not supported out of the box. See Image 2: AWS CodeArtifact. Here, the original package is stored in AWS CodeArtifact and the updated package is stored in the public npm registry. This can cause issues with the integrity of the package and the security of the application.
+
+### Transient Dependencies
+
+Finally, another common issue with Dependabot is the handling of transient dependencies. Transient dependencies are dependencies that are not directly used by the application but are used by the dependencies of the application. A common situation where this becomes a problem is when you want have a large number of PRs, each containing a transient dependency upgrade.
+
+## Solutions
+
+### Notification Fatigue
+
+The first solution to notification fatigue is to limit the number and type of PRs raised by dependabot. Before raising those eyebrows please keep in mind this is exclusive of security updates. Security update PRs from Dependabot in fact can not be limited this config file. I am only talking about patch and minor upgrades being turned off.
+
+To put these limits in place we can create a `.github/dependabot.yml` file in the root of the repository. This file can be used to configure the behavior of Dependabot. Below is an example of a configuration file that limits the number and type of PRs raised by Dependabot.
+
+```yaml
+version: 2
+
+updates:
+  - package-ecosystem: "npm"
+    directory: "/"
+    schedule:
+      interval: "weekly"
+    open-pull-requests-limit: 5
+    commit-message:
+      prefix: "Update"
+      include: "scope"
+    ignore:
+      - dependency-name: "*"
+        update-types:
+          ["version-update:semver-patch", "version-update:semver-minor"]
+```
+
+For a node project in the root directory, this configuration file limits dependabot to 5 open PRs at a time and only allows major updates. Please note this does not include security updates, these are raised irrespective of the `open-pull-requests-limit` parameter. The `ignore` parameter is used to ignore all patch and minor updates.
+
+In addition, prefixing the PRs with `Update` and including the scope of the update in the commit message can help with the readability of the PRs.
+
+With this, at every sprint planning within our team we review the open security and major update PRs of our main repositories, and decide which ones to tackle in the upcoming sprint. However, this in line with our own ways of working and might not be suitable for all teams.
```

Comment:
our own ways of working... at Mechanical Rock?

Reply:
yes